### ISSUE TITLE
You can't mock "Test-Path" 

### ISSUE THREAD

**Issue by DarqueWarrior** [pester/Pester#1551]
## 1. General summary of the issue

I have the need to Mock "Test-Path" if you mock it the tests all fail.

```PowerShell
Tests completed in 104ms
Tests Passed: 0, Failed: 5, Skipped: 0 NotRun: 0
BeforeAll \ AfterAll failed: 1
  - VSTeamProfile
```

## 2. Describe Your Environment
Operating System, Pester version, and PowerShell version:
Pester version     : 5.0.0 /home/dbrown/.local/share/powershell/Modules/Pester/5.0.0/Pester.psd1
PowerShell version : 7.0.1
OS version         : Unix 4.4.0.18362

## 3. Expected Behavior

```PowerShell
Set-StrictMode -Version Latest

Describe 'VSTeamProfile' {
   BeforeAll {
      $sut = (Split-Path -Leaf $PSCommandPath).Replace(".Tests.", ".")

      . "$PSScriptRoot/../../Source/Classes/VSTeamVersions.ps1"
      . "$PSScriptRoot/../../Source/Private/common.ps1"
      . "$PSScriptRoot/../../Source/Public/Get-VSTeamProfile.ps1"
      . "$PSScriptRoot/../../Source/Public/$sut"

      $expectedPath = "$HOME/vsteam_profiles.json"

      Mock Get-Content { return '' }
      Mock Test-Path { return $true }
   }

   Context 'Update-VSTeamProfile entry does not exist' {
      BeforeAll {
         Mock Get-VSTeamProfile { return '[{"Name":"test","URL":"https://dev.azure.com/test","Type":"Pat","Pat":"12345","Version":"VSTS"}]' | ConvertFrom-Json | ForEach-Object { $_ } }
      }

      It 'Should throw' {
         { Update-VSTeamProfile -Name Testing -PersonalAccessToken 678910 } | Should -Throw
      }
   }
}
```

This code should work.
## 4.Current Behavior

```PowerShell
Starting discovery in 1 files.
Discovery finished in 15ms.
[-] Context VSTeamProfile.Update-VSTeamProfile entry does not exist failed
 
Result :
Error :


at Get-AllMockBehaviors, /home/dbrown/.local/share/powershell/Modules/Pester/5.0.0/Pester.psm1: line 11105
at Invoke-Mock, /home/dbrown/.local/share/powershell/Modules/Pester/5.0.0/Pester.psm1: line 11814
at <ScriptBlock><Process>, <No file>: line 91
at <ScriptBlock>, /home/dbrown/.local/share/powershell/Modules/Pester/5.0.0/Pester.psm1: line 12478
at <ScriptBlock>, /home/dbrown/.local/share/powershell/Modules/Pester/5.0.0/Pester.psm1: line 1751
at <ScriptBlock>, /home/dbrown/.local/share/powershell/Modules/Pester/5.0.0/Pester.psm1: line 1730
at Invoke-ScriptBlock, /home/dbrown/.local/share/powershell/Modules/Pester/5.0.0/Pester.psm1: line 1889
at Invoke-Block, /home/dbrown/.local/share/powershell/Modules/Pester/5.0.0/Pester.psm1: line 741
at <ScriptBlock>, /home/dbrown/.local/share/powershell/Modules/Pester/5.0.0/Pester.psm1: line 773
at <ScriptBlock>, /home/dbrown/.local/share/powershell/Modules/Pester/5.0.0/Pester.psm1: line 1764
at <ScriptBlock>, /home/dbrown/.local/share/powershell/Modules/Pester/5.0.0/Pester.psm1: line 1730
at Invoke-ScriptBlock, /home/dbrown/.local/share/powershell/Modules/Pester/5.0.0/Pester.psm1: line 1889
at Invoke-Block, /home/dbrown/.local/share/powershell/Modules/Pester/5.0.0/Pester.psm1: line 785
at Run-Test, /home/dbrown/.local/share/powershell/Modules/Pester/5.0.0/Pester.psm1: line 1451
at Invoke-Test, /home/dbrown/.local/share/powershell/Modules/Pester/5.0.0/Pester.psm1: line 2226
at Invoke-Pester<End>, /home/dbrown/.local/share/powershell/Modules/Pester/5.0.0/Pester.psm1: line 3995
at <ScriptBlock>, /home/dbrown/.vscode-server/extensions/ms-vscode.powershell-2020.4.0/InvokePesterStub.ps1: line 125
at <ScriptBlock>, <No file>: line 1

[-] Describe VSTeamProfile failed
 RuntimeException: Framework failed
 
Tests completed in 98ms
Tests Passed: 0, Failed: 1, Skipped: 0 NotRun: 0
BeforeAll \ AfterAll failed: 1
  - VSTeamProfile
```
## 5. Possible Solution

If you comment out the Mock for Test-Path the code works. But if I am on a clean system where Test-Path would return false this test will fail. I need to be able to mock Test-Path

**Comment by nohwnd** [comment]
Please run this with the Diagnostic output so I can see what is happening. `Invoke-Pester -Output Diagnostic`

**Comment by chridentitv** [comment]
Script to reproduce:
```
Describe Do-Something {
    BeforeAll {
        function Do-Something { }

        Mock Test-Path { $true }
    }

    Context 'Some block' {
        It 'does something' {
            Do-Something
        }
    }
}
```
Attaching diagnostic output.
```
Discovery: Starting test discovery in 1 test containers.

Starting discovery in 1 files.
Discovery: Discovering tests in C:\temp\test.tests.ps1
Discovering in C:\temp\test.tests.ps1.
Found 1 tests. 16ms
Discovery: Found 1 tests in 26 ms
Discovery: Processing discovery result objects, to set root, parents, filters etc.
Filter: (Do-Something) Block will be included in the run, because there were no include filters, and will let its children to determine whether or not it should run.
Filter: (Do-Something.Some block) Block will be included in the run, because there were no include filters, and will let its children to determine whether or not it should run.
Filter: (Do-Something.Some block.does something) Test will be included in the run, because there were no include filters so all tests are included unless they match exclude rule.
Discovery finished in 61ms.
Discovery: Test discovery finished.

Running tests from 'C:\temp\test.tests.ps1'
Describing Do-Something
Mock: Setting up default mock for Test-Path.
Mock: We are in a block, one time setup or similar. Returning mock table from test block.
Mock: Resolving command Test-Path.
Mock: Searching for command Test-Path in the caller scope.
Mock: Found the command Test-Path in the caller scope.
Mock: Mock does not have a hook yet, creating a new one.
Mock: Defined new hook with bootstrap function PesterMock_09c26d4a-28c6-4ad7-bcaf-59bbb45ceb8a and aliases Test-Path, Microsoft.PowerShell.Management\Test-Path.
Mock: Adding a new default behavior to Test-Path.
 Context Some block
Mock: Mock for Test-Path was invoked from block Process, resolving call history and behaviors.
Mock: Getting all defined mock behaviors in this and parent scopes for command Test-Path.
Mock: Finding all behaviors in this block and parent blocks.
[-] Context Do-Something.Some block failed

Result :
Error :
Mock: Mock for Test-Path was invoked from block Process, resolving call history and behaviors.
Mock: Getting all defined mock behaviors in this and parent scopes for command Test-Path.
Mock: Finding all behaviors in this block and parent blocks.


at Get-AllMockBehaviors, C:\Users\Chris\Documents\PowerShell\Modules\Pester\5.0.1\Pester.psm1: line 11269
at Invoke-Mock, C:\Users\Chris\Documents\PowerShell\Modules\Pester\5.0.1\Pester.psm1: line 11983
at <ScriptBlock><Process>, <No file>: line 91
at <ScriptBlock>, C:\Users\Chris\Documents\PowerShell\Modules\Pester\5.0.1\Pester.psm1: line 12647
at <ScriptBlock>, C:\Users\Chris\Documents\PowerShell\Modules\Pester\5.0.1\Pester.psm1: line 1751
at <ScriptBlock>, C:\Users\Chris\Documents\PowerShell\Modules\Pester\5.0.1\Pester.psm1: line 1730
at Invoke-ScriptBlock, C:\Users\Chris\Documents\PowerShell\Modules\Pester\5.0.1\Pester.psm1: line 1889
at Invoke-Block, C:\Users\Chris\Documents\PowerShell\Modules\Pester\5.0.1\Pester.psm1: line 741
at <ScriptBlock>, C:\Users\Chris\Documents\PowerShell\Modules\Pester\5.0.1\Pester.psm1: line 773
at <ScriptBlock>, C:\Users\Chris\Documents\PowerShell\Modules\Pester\5.0.1\Pester.psm1: line 1764
at <ScriptBlock>, C:\Users\Chris\Documents\PowerShell\Modules\Pester\5.0.1\Pester.psm1: line 1730
at Invoke-ScriptBlock, C:\Users\Chris\Documents\PowerShell\Modules\Pester\5.0.1\Pester.psm1: line 1889
at Invoke-Block, C:\Users\Chris\Documents\PowerShell\Modules\Pester\5.0.1\Pester.psm1: line 785
at Run-Test, C:\Users\Chris\Documents\PowerShell\Modules\Pester\5.0.1\Pester.psm1: line 1451
at Invoke-Test, C:\Users\Chris\Documents\PowerShell\Modules\Pester\5.0.1\Pester.psm1: line 2226
at Invoke-Pester<End>, C:\Users\Chris\Documents\PowerShell\Modules\Pester\5.0.1\Pester.psm1: line 4201
at <ScriptBlock>, <No file>: line 1

Mock: Removing function PesterMock_09c26d4a-28c6-4ad7-bcaf-59bbb45ceb8a and aliases Test-Path, Microsoft.PowerShell.Management\Test-Path for Test-Path.
[-] Describe Do-Something failed
 RuntimeException: Framework failed

Tests completed in 307ms
Tests Passed: 0, Failed: 1, Skipped: 0 NotRun: 0
BeforeAll \ AfterAll failed: 1
  - Do-Something
```

**Comment by nohwnd** [comment]
Thanks for the simple repro. I forgot to use SafeCommands in all the places, this should be fixed and will be released in 5.0.2 this evening.
